tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -376,8 +376,10 @@ ROOT::Experimental::Detail::RPage ROOT::Experimental::Detail::RPageSourceFile::P
       R__ASSERT(fCurrentCluster->ContainsColumn(columnId));
 
       auto cachedPage = fPagePool->GetPage(columnId, RClusterIndex(clusterId, idxInCluster));
-      if (!cachedPage.IsNull())
+      if (!cachedPage.IsNull()) {
+         delete[] pageBuffer;
```

Comment:
Could we add a comment (probably around line 406) about what entity takes ownership of the buffer when it is not deleted?